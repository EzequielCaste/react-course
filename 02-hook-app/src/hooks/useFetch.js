import React, { useState, useEffect } from 'react'


export const useFetch = ( url ) => {

  const [state, setState] = useState({ data: null, loading: true, error: null })

  useEffect( () => {

    setState({ loading: true, data: null, error: null })
    fetch( url )
      .then( resp => resp.json() )
      .then( data => {

        setState({
          loading: false,
          error: null,
          data: data
        })
      })
  }, [url]);

  return state;
}
