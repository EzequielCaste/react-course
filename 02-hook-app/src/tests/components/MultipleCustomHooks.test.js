import React from 'react';
import { shallow } from 'enzyme';
import { MultipleCustomHooks } from '../../components/03-ejemplos/MultipleCustomHooks';
import { useFetch } from '../../hooks/useFetch';
import { useCounter } from '../../hooks/useCounter';
jest.mock('../../hooks/useFetch');
jest.mock('../../hooks/useCounter');



describe('Pruebas en <MultipleCustomHooks /> ', () => {

  useCounter.mockReturnValue({
    counter: 10,
    increment: () => {}
  })
 
  test('debe mostrarse correctamente', () => {

    useFetch.mockReturnValue({
      data: null,
      loading: false,
      error: null
    });

    const wrapper = shallow(<MultipleCustomHooks />);
    expect( wrapper ).toMatchSnapshot();


  })

  test('debe mostrar la informacion', () => {
    useFetch.mockReturnValue({
      data: [{
        author: 'Ezequiel',
        quote: 'Hola Mundo'
      }],
      loading: false,
      error: null
    });
    const wrapper = shallow(<MultipleCustomHooks />);

    expect( wrapper.find('.alert').exists() ).toBe(false);
    expect( wrapper.find('footer').text().trim() ).toBe("Ezequiel")
    expect( wrapper.find('.mb-0').text().trim() ).toBe("Hola Mundo")

    
  })


})