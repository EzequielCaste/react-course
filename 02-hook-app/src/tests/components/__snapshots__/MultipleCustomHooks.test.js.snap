// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <MultipleCustomHooks />  debe mostrarse correctamente 1`] = `
<div>
  <h1>
    Breaking Bad Quotes
  </h1>
  <hr />
  <blockquote
    className="blockquote text-right"
  >
    <p
      className="mb-0"
    />
    <footer
      className="blockquote-footer"
    >
       
    </footer>
  </blockquote>
  <button
    className="btn btn-primary"
    onClick={[Function]}
  >
    Next Quote
  </button>
</div>
`;
